test/build/base-url-homepage/base-url-homepage.test.js

```diff
@@ -7,10 +7,10 @@ const cwd = path.join(__dirname, 'build');
 let files = {};
 
 describe('packageManifest.homepage', () => {
-  beforeAll(() => {
-    setupBuildTest(__dirname);
+  beforeAll(async () => {
+    await setupBuildTest(__dirname);
 
-    files = readFiles(['index.html'], {cwd});
+    files = await readFiles(cwd);
```

Comment:
…as is `readFiles()` (and it was changed to take an entire directory, and read UTF-8-friendly files)

Reply:
Unfortunately, I had to revert the “async” setup. Though it was working locally, it was consistently failing in CI.

I fought with it for a couple hours, trying to figure out what the issue is. I never got to the bottom of it, but I suspect [async weirdness in Jest](https://github.com/facebook/jest/issues/8688). Reason is that `beforeAll(async()` was never originally supported; you’ll see older issues demanding this feature. Then when it was added, I’m seeing _very_ weird behavior that I can’t really explain other than guessing that `beforeAll(async()` doesn’t use proper Node.js promises. I can’t explain otherwise why it just would never truly wait for async functions to finish (I tried everything I could think of 😅)

But that said, I **don’t** think that having synchronous setup functions is necessarily slower. Jest spawns each individual test in its own child process on its own thread, so already this work is being parallelized. If my suspicions are correct, then I think the sync processes here are happening within the context of a child process, so it’s already optimized.

But also, I’m scared with this much difficulty of me or someone else fighting with `beforeAll(async()` weirdness, so I’d rather just avoid it.